packages/create-redwood-app/template/api/src/lib/prisma.ts

```diff
@@ -0,0 +1,91 @@
+import { PrismaClient, Prisma } from '@prisma/client'
+
+import { logger } from './logger'
+
+// To help you identify particularly slow parameter values,
+// For example, Heroku outputs the slowest queries (that take 2 seconds or more)
+const SLOW_QUERY_THRESHOLD = 2 * 1000 // 2 seconds
+
+/*
+ * Determines the type and level of logging.
+ *
+ * See: https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#log
+ */
+export const defaultLogLevels: Prisma.LogLevel[] = ['info', 'warn', 'error']
+
+/*
+ * Generates the Prisma Log Definitions from log levels
+ *
+ * @return Prisma.LogDefinition[]
+ */
+export const prismaLoggerOptions = (
+  logLevels: Prisma.LogLevel[]
+): Prisma.LogDefinition[] => {
+  return logLevels?.map((level) => {
+    return { emit: 'event', level } as Prisma.LogDefinition
+  })
+}
```

Comment:
Can we hide these implementations in the logger library in the api package?

I'm concerned that Prisma changes the format and then we've shipped a bunch of this configuration to end-users.

Reply:
This, + setLogLevels

Reply:
> Can we hide these implementations in the logger library in the api package?

@peterp I didn't know if I wanted to have the api/logger depend on Prisma.

Some deploys won't have Prisma so then would that prevent people from using the logger?

> I'm concerned that Prisma changes the format 

I would actually like them to -- it is cumbersome to have to set these when init-ing a client. You cannot set them after the fact.

But not sure I can think of an alternative.

Reply:
@peterp 

> This, + setLogLevels

Sorry, I don't follow. Rename `logLevels` to `setLogLevels`?

It is used

```js
/*
 * Generates the Prisma Log Definitions from log levels
 *
 * @return PrismaClient
 */
export const createPrismaClient = (logLevels?: Prisma.LogLevel[]) => {
  const client = new PrismaClient({
    log: prismaLoggerOptions(logLevels),
    errorFormat: 'colorless',
  })

  logLevels && setLogLevels(client, logLevels)

  return client
}
```

I can do that, thenI would just need to rename:

`setLogLevels(client, logLevels)` to something like `configureLogLevels` or `defineLogLevels`? 

Or since they are different types, can I use the same name? I'll see.

Reply:
No, I didn't think I could:

![image](https://user-images.githubusercontent.com/1051633/111640437-935a9080-87d2-11eb-8e0e-fcdf7fc1f9a0.png)